src/gui/dragAndDrop.js

```diff
@@ -27,6 +28,7 @@ function dropNode(event) {
         clone.style.top = `${event.pageY}px`;
         clone.style.left = `${event.pageX}px`;
         event.target.appendChild(clone);
+        NodeForest.insertRootNode(nodeCount);
```

Comment:
Adding nodes to the forest is already implemented by Us4.

Reply:
I have merged this with mine, have another look